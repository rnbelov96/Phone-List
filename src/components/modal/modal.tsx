import React from 'react';
import { connect } from 'react-redux';
import { AppActionCreators } from '../../reducer/app/app';

export type ModalProps = {
  photoUrl: string;

  setModalWindowUrl: (url: string | null) => void;
};

export const PureModal: React.FunctionComponent<ModalProps> = ({
  photoUrl,
  setModalWindowUrl,
}: ModalProps) => {
  const [isImageLoading, setImageLoadingStatus] = React.useState<boolean>(true);
  return (
    <div className="modal-window">
      <div
        className="modal-window__center-wrapper"
        onClick={e => {
          if (e.target === e.currentTarget) {
            setModalWindowUrl(null);
            document.body.style.overflowY = 'auto';
            document.body.style.paddingRight = '0px';
          }
        }}
      >
        <img
          style={{
            display: isImageLoading ? 'none' : 'inline-block',
          }}
          onLoad={() => setImageLoadingStatus(false)}
          src={photoUrl}
          alt="big square"
        />
        <div
          style={{
            display: isImageLoading ? 'inline-block' : 'none',
          }}
          className="lds-ring mx-auto"
        >
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  );
};

export default connect(() => ({}), {
  setModalWindowUrl: AppActionCreators.setModalWindowUrl,
})(PureModal);
