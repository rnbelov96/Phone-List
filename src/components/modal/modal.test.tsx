import React from 'react';
import renderer from 'react-test-renderer';
import { PureModal, ModalProps } from './modal';

const mockProps: ModalProps = {
  setModalWindowUrl: jest.fn(),
  photoUrl: 'some url',
};

test('Modal should render correctly', () => {
  const tree = renderer.create(<PureModal {...mockProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
