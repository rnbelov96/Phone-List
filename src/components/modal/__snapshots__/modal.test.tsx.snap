// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
<div
  className="modal-window"
>
  <div
    className="modal-window__center-wrapper"
    onClick={[Function]}
  >
    <img
      alt="big square"
      onLoad={[Function]}
      src="some url"
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div
      className="lds-ring mx-auto"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
